string type
float64 batteryLevel
GPS gps
string serialNumber
string messageType 

